user_guide_src/source/incoming/controllers.rst

```diff
@@ -9,9 +9,118 @@ Controllers are the heart of your application, as they determine how HTTP reques
     :depth: 2
 
 What is a Controller?
+*********************
+
+A Controller is simply a class file that handles a HTTP request. :doc:`URI Routing <routing>` associates a URI with a controller.
+
+Included Properties
+*******************
+
+Every controller you create should extend ``CodeIgniter\Controller`` class.
+This class provides several features that are available to all of your controllers.
+
+**Request Object**
+
+The application's main :doc:`Request Instance </incoming/incomingrequest>` is always available
+as a class property, ``$this->request``.
+
+**Response Object**
+
+The application's main :doc:`Response Instance </outgoing/response>` is always available
+as a class property, ``$this->response``.
+
+**Logger Object**
+
+An instance of the :doc:`Logger <../general/logging>` class is available as a class property,
+``$this->logger``.
+
+Helpers
+=======
+
+You can define an array of helper files as a class property. Whenever the controller is loaded
+these helper files will be automatically loaded into memory so that you can use their methods anywhere
+inside the controller:
+
+.. literalinclude:: controllers/016.php
+   :lines: 2-
```

Comment:
Why did this come back? Was it a rebase?

Reply:
I've started this work before your `:lines:` deletions.
Today I reworked this file, and pasted the old file into it to check the difference.
And I deleted `:lines:` lines, but it seems I forgot to delete the line.

Reply:
Fixed and rebased.

Reply:
👍